Improved Profile Picture UI

yougotwill opened the pull request:
- Converted EditProfileDialog to a functional component.
- Added a modal to edit the user's profile picture
  - Supports updating and removing the profile picture
- Added proper typings to icons and added thumbnail icon
- Updated profile picture integration tests
- Sort localised keys alphabetically

yougotwill committed:
feat: convered EditProfileDialog to a functional component

yougotwill committed:
feat: initial work for set display picture modal done

still need to do logic and handle multiple modals on screen

yougotwill committed:
feat: added button state logic and migrated avatar upload logic from EditProfileDialog

show loading spinner while avatar is loading

yougotwill committed:
feat: initial remove profile work done

yougotwill committed:
feat: small refactoring

yougotwill committed:
feat: desktop now supports avatar being removed via sync configuration message

disabled buttons while loading

yougotwill committed:
feat: added proper typings to icons and added thumbnail icon

confirmed support for svgs with multiple paths

yougotwill committed:
feat: added upload image button if user doesn't have an avatar set

yougotwill committed:
fix: moved all avatar logic from editProfileDialog to DisplayPictureModal

yougotwill committed:
fix: rename oldAvatarPath to avatarPath for clarity

yougotwill committed:
refactor: rename DisplayPictureModal to EditProfilePictureModal

yougotwill committed:
fix: pass yarn ready

yougotwill committed:
fix: cleanup

yougotwill committed:
fix: EditProfilePictureModalState should be camel case

yougotwill commented on ts/components/dialog/ModalContainer.tsx:
```diff
@@ -55,6 +57,7 @@ export const ModalContainer = () => {
   const banOrUnbanUserModalState = useSelector(getBanOrUnbanUserModalState);
   const reactListModalState = useSelector(getReactListDialog);
   const reactClearAllModalState = useSelector(getReactClearAllDialog);
+  const EditProfilePictureModalState = useSelector(getEditProfilePictureModalState);
```
Fixed

yougotwill commented on ts/components/icon/Icons.tsx:
```diff
@@ -459,6 +463,16 @@ export const icons = {
     viewBox: '-1 -1 35 35',
     ratio: 1,
   },
+  thumbnail: {
+    path: [
+      'M34.915 23.8987L26.594 17.812C26.5393 17.7682 26.4714 17.7444 26.4014 17.7444C26.3313 17.7444 26.2634 17.7682 26.2087 17.812L18.748 23.9886C18.6905 24.0313 18.6207 24.0544 18.549 24.0544C18.4773 24.0544 18.4075 24.0313 18.35 23.9886L14.6003 20.9324C14.5468 20.8918 14.4814 20.8698 14.4141 20.8698C14.3469 20.8698 14.2815 20.8918 14.2279 20.9324L4.0706 28.1491C4.03144 28.1797 3.99959 28.2186  ...
+      'M11.0947 18.7492C12.6904 18.7492 13.984 17.4556 13.984 15.8599C13.984 14.2642 12.6904 12.9707 11.0947 12.9707C9.49902 12.9707 8.20546 14.2642 8.20546 15.8599C8.20546 17.4556 9.49902 18.7492 11.0947 18.7492Z',
+      'M40.3597 4.62408L8.19261 0.874467C7.77128 0.822682 7.34385 0.855312 6.93527 0.970455C6.52668 1.0856 6.14511 1.28095 5.81282 1.54511C5.48052 1.80927 5.20415 2.13696 4.99984 2.50905C4.79552 2.88115 4.66734 3.2902 4.62277 3.71236L4.46868 5.0992H7.03691L7.16532 4.00771C7.17565 3.92437 7.20256 3.84394 7.24446 3.77116C7.28637 3.69838 7.34241 3.63473 7.4093 3.58395C7.51881 3.496 7.65413 3.44638 7 ...
+      'M35.6855 35.3016H3.30013C2.44871 35.3016 1.63216 34.9634 1.03011 34.3613C0.428069 33.7593 0.0898438 32.9427 0.0898438 32.0913V10.2614C0.0898438 9.40994 0.428069 8.59339 1.03011 7.99135C1.63216 7.3893 2.44871 7.05108 3.30013 7.05108H35.6855C36.5369 7.05108 37.3535 7.3893 37.9555 7.99135C38.5576 8.59339 38.8958 9.40994 38.8958 10.2614V32.0913C38.8958 32.9427 38.5576 33.7593 37.9555 34.3613C3 ...
+    ],
+    viewBox: '0 0 44 36',
+    ratio: 1,
+  },
```
Very nice! Will update. Fixed

yougotwill committed:
fix: compressed and flattened thumbail svg path

yougotwill commented on ts/components/dialog/EditProfileDialog.tsx:
```diff
@@ -51,310 +48,255 @@ const QRView = ({ sessionID }: { sessionID: string }) => {
+                text={window.i18n('save')}
+                buttonType={SessionButtonType.Simple}
+                onClick={onClickOK}
+                disabled={loading}
```
True, it was there before I decided to hide the buttons

yougotwill commented on ts/receiver/userProfileImageUpdates.ts:
```diff
@@ -139,6 +141,11 @@ async function createOrUpdateProfile(
         // do not return here, we still want to update the display name even if the avatar failed to download
       }
     }
+  } else if (ourProfile && profileKey) {
+    // Note: this is a special case where we are updating our own profile, and we have a profile key (which is probably empty) and no profile image - this means we need to clear our avatar
+    await clearOurAvatar(false);
+
+    changes = true;
```
Doesn't the profile key need to be defined so that we know there is an avatar that needs to be cleared?

Bilb replied:
yeah right. I think that works then

yougotwill committed:
fix: sort localised keys generated by updateI18nKeysType

yougotwill commented on ts/types/LocalizerKeys.ts:
```diff
@@ -1,501 +1,502 @@
+  | 'readReceiptSettingDescription'
+  | 'readReceiptSettingTitle'
+  | 'typingIndicatorsSettingDescription'
+  | 'typingIndicatorsSettingTitle'
```
Keys are now sorted alphabetically. Minimum python version is 3.6 to use Ordered Dictionaries

burtonemily commented:
Tested and approved :)

yougotwill committed:
fix: icon path typing error

yougotwill committed:
fix: removed duplicate imports and ordered localised keys

Bilb committed:
Merge pull request #2824 from oxen-io/unstable

Merge unstable to clearnet with userconfig

Bilb committed:
Merge pull request #2825 from oxen-io/clearnet

Merge userconfig scheduled release

yougotwill committed:
feat: fixed integration tests

except for linked device profile sync, need to update avatar-update-blue files for linux

Bilb committed:
test: fix update profile picture snapshots

- loop until match when validating screenshot
- wait 15s and take screenshot when updating screenshot

yougotwill committed:
Merge pull request #31 from Bilb/feature/ses-476/remove-profile-picture

test: fix update profile picture snapshots

yougotwill committed:
fix: updated avatar to correct image

burtonemily commented on ts/test/automation/user_actions.spec.ts:
Is there a specific reason to change the test name here? If there is I will update the mobile platforms but otherwise we should keep the tests name unchanged.

yougotwill replied:
The reason is so that we can update all profile picture snapshots through one command.

If we do `yarn run integration-test-snapshots` it does `npx playwright test -g 'profile picture' --update-snapshots"` which will run the "Check profile picture syncs" test and create a new snapshot. The "Change avatar" test had to be run manually. By making it profile picture both tests run off the same command. 

So it's more of a convenience thing. Happy to change it back if it is going to cause problems?

burtonemily replied:
So its updating both the linked device test and the change avatar test in one go, if you change the name of the change avatar test? Is it a script that you've created or is this playwright functionality?

yougotwill replied:
It's using the "-g" flag it will look for any tests that contain the string 'profile picture' and only those 2 tests match. No additional scripts have been added.

burtonemily replied:
Cool, will update the appium tests to match👌

yougotwill committed:
fix: revert rm to rmdirSync so we don't remove config files before a test has been completed

Bilb commented:
Closing this inf favor if https://github.com/oxen-io/session-desktop/pull/2869
